presto-spi/src/main/java/com/facebook/presto/spi/type/Decimals.java

```diff
@@ -232,6 +234,14 @@ public static void checkOverflow(BigInteger value)
         }
     }
 
+    public static BigDecimal readBigDecimal(DecimalType type, Block block, int position)
+    {
+        BigInteger unscaledValue = type.isShort()
+                ? BigInteger.valueOf(type.getLong(block, position))
+                : decodeUnscaledValue(type.getSlice(block, position));
+        return new BigDecimal(unscaledValue, type.getScale(), new MathContext(type.getPrecision()));
```

Comment:
It would be nice if this commit also updated:

- `com.facebook.presto.spi.RecordPageSink#decodeDecimalValue(int, com.facebook.presto.spi.block.Block, com.facebook.presto.spi.type.DecimalType)`
- `com.facebook.presto.mongodb.MongoPageSink#getObjectValue` (although this one looks weird.. it constructs `BigDecimal` from unscaled value but doesn't pass scale there..)

Reply:
Will update

Reply:
I'm not going to touch the MongoDB one since that seems to have different (wrong?) behavior and doesn't appear to have tests. I avoided changing `RecordPageSink` since it gets deleted in a subsequent commit. But updating it seems like a good idea, since it shows that this method is equivalent and runs the tests.

Reply:
Lack of tests for MongoDB might be discouraging.. Maybe leave a TODO/FIXME there.

Reply:
Good idea! Will add a TODO